### PR TITLE
Skip roll dialog

### PR DESCRIPTION
**Pull request by IceFreez3r**
- Added the option to pass `skipDialog` to a skill or attack roll, to skip the dialog asking for parameters. Instead the roll is immediately executed with default parameters
- Converted checkDialog to TS
- Added `rollModes` to `foundryApi`.
- Added new integration as `splittermond.apiConstants` that checks the above.
- Improved documentation of integration tests in README

~~TODO: Test for skipping the check dialog~~ Done

### CODE REVIEW

File: src/module/apps/dialog/check-dialog.ts [hunk 1]
```diff
@@ -1,9 +1,37 @@
+import Skill from "module/actor/skill";
 import { foundryApi } from "../../api/foundryApi";
 import { FoundryDialog } from "module/api/Application";
+import { RollMode } from "module/api/foundryTypes";
+import { ApplicationRenderContext } from "module/data/SplittermondApplication";
+import { RollType } from "module/config/RollType";
+
+export type RollDifficultyString = number | "VTD" | "KW" | "GW";
+
+export interface CheckDialogProps {
```
Comment (m-2squared) [inline src/module/apps/dialog/check-dialog.ts]: I would prefer `CheckDialogInput`

File: README.md [hunk 2]
```diff
@@ -145,9 +145,13 @@ auch eine Reihe globaler Mocks für die Tests:
 
 #### Integration Tests
 
-Integrationstests werden mithilfe des Moduls `@ethaks/fvtt-quench`. Diese Tests konzentrieren sich auf das Testen der
-Foundry API (so sie von uns benutzt wird), sowie tests die viele Foundry Komponenten, insbesondere GUI Komponenten,
-involvieren. Sie sind so gebaut, dass in einer frischen Splittermond Welt funktionieren.
+Integration Tests konzentrieren sich auf das Testen der Foundry API (so sie von uns benutzt wird), sowie tests die viele
```
Comment (m-2squared) [inline README.md]: Improved documentation? Nice!

**Review by m-2squared — CHANGES_REQUESTED** [review]
Thanks for your effort, especially in adding tests and migrating to TS. 
I have some minor points though, that I'd like to be addressed.

Don't forget to add yourself a CHANGELOG entry.

File: src/__tests__/unit/module/actor/attack.test.ts [hunk 3]
```diff
@@ -345,6 +347,37 @@ describe("Attack", () => {
...
+            roll: {
+                dice: [],
+                total: 26,
+                getTooltip: () => "",
+            },
+        });
+        sandbox.stub(Chat, "prepareCheckMessageData").resolves();
+        sandbox.stub(foundryApi, "createChatMessage").resolves();
+
+        await underTest.roll(customOptions);
```
[the remark this answers is not in the document]
Comment (l-1squared) [reply src/__tests__/unit/module/actor/attack.test.ts]: yes please

File: src/__tests__/unit/module/actor/skill.test.ts [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+import SplittermondActor from "module/actor/actor";
+import { CharacterDataModel } from "module/actor/dataModel/CharacterDataModel";
+import ModifierManager from "module/actor/modifier-manager";
+import Skill from "module/actor/skill";
+import { foundryApi } from "module/api/foundryApi";
+import { Dice } from "module/util/dice";
+import { Chat } from "module/util/chat";
+import CheckDialog from "module/apps/dialog/check-dialog";
+
+describe("Skill", () => {
```
Comment (l-1squared) [inline src/__tests__/unit/module/actor/skill.test.ts]: How about a third test case that this still works if you don't ask the user but forget to provide input.
There is actually another edge case where a user provides absolute garbage thati s not falsy. But I'm willing to have that fail deep in the check logic

File: src/__tests__/unit/module/actor/skill.test.ts [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+        sandbox.stub(foundryApi, "localize").callsFake((key: string) => key);
+        stubRollApi(sandbox);
+    });
+
+    afterEach(() => sandbox.restore());
+
+    it("should make a skill check with the options from the check dialog", async () => {
+        const actor = setUpActor(sandbox);
+        const underTest = new Skill(actor, "Testskill");
+        CheckDialog.create;
```
Comment (m-2squared) [inline src/__tests__/unit/module/actor/skill.test.ts]: looks useless.

File: src/__tests__/unit/module/actor/skill.test.ts (hunk 4 above)
Comment (m-2squared) [inline src/__tests__/unit/module/actor/skill.test.ts]: I think there is a third test case where the user forgets to provide input but has askUser=false.

Then there another edge case where the user provides non-falsy garbage. But that's on them :)

Comment (IceFreez3r) [reply src/__tests__/unit/module/actor/skill.test.ts]: If we assume that the types of the input are followed (even though `Skill` hasn't been transformed yet), then every input either has the expected type or is nullish. I could remove all the options from the second test, since they are all the default, but in practice the JSDoc `{?number}` gets interpreted as `number | null` and TS doesn't allow undefined. So I could add a test where I just pass `{ askUser: false }`, but then I would need to silence TS in the test.

File: src/__tests__/unit/module/actor/skill.test.ts (hunk 5 above)
Comment (IceFreez3r) [reply src/__tests__/unit/module/actor/skill.test.ts]: removed
